polkadot/runtime/parachains/src/paras_inherent/tests.rs

```diff
@@ -3778,6 +3788,255 @@ mod sanitizers {
+							.map(|(i, v)| (*i, validators_to_keyrings[v]))
+							.collect();
+
+						make_dispute_concluding_against(c.hash(), session, &keyrings)
+					})
+					.collect();
+
+				let parent_block_number = RELAY_PARENT_NUMBER;
+				let parent_header = HeaderFor::<Test>::new(
+					parent_block_number,     // `block_number`,
+					Default::default(),      // `extrinsics_root`,
+					Default::default(),      // `storage_root`,
+					default_header().hash(), // `parent_hash`,
+					Default::default(),      // digest,
+				);
+
+				run_to_block(parent_block_number + 1, |_| None);
+				frame_system::Pallet::<Test>::reset_events();
+				frame_system::Pallet::<Test>::initialize(
+					&(parent_header.number() + 1),
+					&parent_header.hash(),
+					&Digest { logs: Vec::new() },
+				);
+
+				let n_disputes = disputes.len();
+				let data = ParachainsInherentData {
+					disputes,
+					parent_header: parent_header.clone(),
+					bitfields: Vec::new(),
+					backed_candidates: Vec::new(),
+				};
+				let mut inherent_data = InherentData::new();
+				inherent_data.put_data(PARACHAINS_INHERENT_IDENTIFIER, &data).unwrap();
+
+				let header = default_header();
+				shared::Pallet::<Test>::add_allowed_relay_parent(
+					header.hash(),
+					*header.state_root(),
+					RELAY_PARENT_NUMBER,
+					10,
+				);
+				// Update scheduler's claimqueue with the parachains
+				let ttl = RELAY_PARENT_NUMBER + 10;
+				scheduler::Pallet::<Test>::set_claim_queue(BTreeMap::from([
+					(
+						CoreIndex::from(0),
+						VecDeque::from([ParasEntry::new(
+							Assignment::Pool { para_id: 1.into(), core_index: CoreIndex(0) },
+							ttl,
+						)]),
+					),
+					(
+						CoreIndex::from(1),
+						VecDeque::from([ParasEntry::new(
+							Assignment::Pool { para_id: 2.into(), core_index: CoreIndex(1) },
+							ttl,
+						)]),
+					),
+				]));
+				let _ = Pallet::<Test>::enter(frame_system::RawOrigin::None.into(), data).unwrap();
+
+				assert_matches!(pallet::OnChainVotes::<Test>::get(), Some(ScrapedOnChainVotes {
+					disputes,
+					..
+				} ) => {
+					assert_eq!(disputes.len(), n_disputes);
+				});
+
+				let parent_block_number = RELAY_PARENT_NUMBER + 2;
+				run_to_block(parent_block_number, |_| None);
+				let parent_header = HeaderFor::<Test>::new(
+					parent_block_number,  // `block_number`,
+					Default::default(),   // `extrinsics_root`,
+					Default::default(),   // `storage_root`,
+					parent_header.hash(), // `parent_hash`,
+					Default::default(),   // digest,
+				);
+
+				frame_system::Pallet::<Test>::reset_events();
+				frame_system::Pallet::<Test>::initialize(
+					&(parent_header.number() + 1),
+					&parent_header.hash(),
+					&Digest { logs: Vec::new() },
+				);
+
+				let data = ParachainsInherentData {
+					disputes: Vec::new(),
+					parent_header,
+					bitfields: Vec::new(),
+					backed_candidates,
+				};
+
+				let _ =
+					Pallet::<Test>::enter(frame_system::RawOrigin::None.into(), data).unwrap_err();
```

Comment:
we should match the error here. I assume it must be a CandidatesFilteredDuringExecution, right?
maybe we should also call explicitly sanitize_backed_candidates and validate the output